In [2]:
# PP6
# Group members: Gabin Ryu, David Assaraf, Yang Xiang


![](graph.png)

In [15]:
# Computational graph of function f

import numpy as np
def f(x, y):
    return np.exp(-(np.sin(x)-np.cos(y))**2)

print(f(np.pi/2, np.pi/3))

0.778800783071405


| Trace    | Elementary Operation  | Value of Elementary function   | Derivative             | Derivative Value with respect to $x_{1}$ | Derivative Value with respect to $x_{2}$    
| :------: | :----------------------:| :----: |   :--------------------:         | :------------------------------: | : -----------------------------------: |
| $x_{1}$  | $\dfrac{\pi}{2}$   | $1.571$ |           $1$      |         $1$                 |    $0$             |
| $x_{2}$  | $\dfrac{\pi}{3}$  | $1.047$ |        $1$          |          $0$               |        $1$                          |
| $v_{1}$  | $sin(x_{1})$      |$1.000$  |        $cos(x_{1})\dot{x_{1}}$       |    $0$                      |        $0$                |
| $v_{2}$  | $cos(x_{2})$      | $0.500$  |   $-sin(x_{2})\dot{x_{2}}$    |           $0$            |   $-0.866$                        |
| $v_{3}$  | $v_{1}-v_{2}$   |  $-0.500$   |      $\dot{v_{1}} -\dot{v_{2}}$           |     $0$                    |    $0.866$                  |
| $v_{4}$  | $v_{3}^2$       |  $0.250$  |        $2v_{3}\dot{v_{3}}$           |       $0$                  |              $-0.866$       |
| $v_{5}$  | $-v_{4}$       | $-0.250$     | $-\dot{v_{4}}$    |             $0$             |       $0.866$                      |
| $v_{6}$  | $exp(v_{5})$    |   $0.779$   |        $\dot{v_{5}}exp(v_{5})$              |       $0$                 |         $0.674$                |



$f(\frac{\pi}{2},\frac{\pi}{3})$ = 0.779

$\frac{\partial f}{\partial x}(\frac{\pi}{2},\frac{\pi}{3})$ = 0


$\frac{\partial f}{\partial y}(\frac{\pi}{2},\frac{\pi}{3})$ = 0.674
